crates/bevy_reflect/src/tuple.rs

```diff
@@ -400,6 +465,16 @@ macro_rules! impl_reflect_tuple {
             }
         }
 
+        impl <$($name: Typed),*> Typed for ($($name,)*) {
+            fn type_info() -> TypeInfo {
+                let fields = [
+                    $(UnnamedField::new::<$name>($index),)*
+                ];
```

Comment:
Can we make this `const` in some way? Would cut down on runtime allocations? Even with caching, this could reduce the size of the *Info structs.

Reply:
I think so. Could we make it a `&'static [UnnamedField]` type? The problem with the tuple case is that we can't get the number of args from the macro_rules. I guess if we need to, we could just add a count parameter to the macro.

Reply:
Okay, I don't think this can be `const` due to `TypeId::of::<T>()` itself not being `const`.